File: tests/run/t12300/B_2.java
class OneTwo implements One, Two {}
// Java doesn't seen class parents of traits
class YSubX extends Y implements YSub {}

public class B_2 {
  public static void foo() {
    A a = new A();
    a.foo1(new One() {});
    a.foo2(new One() {});
    a.foo3(new OneTwo());
    a.foo4(new OneTwo());
    a.foo5(new Y());
    a.foo6(new Y());
    a.foo7(new YSubX());
    a.foo8(new YSubX());
  }
}

File: tests/run/i18884/B_2.scala
import lib.*

@main def Test: Unit =
  test(new Foo1(1))
  test(new Foo2(2, 3))
  test(new Foo3(4, 5))
  test(new Foo4(6, 7))

def test(any: Any): Unit =
  any match
    case Foo1(x) => println(s"Foo1($x)")
    case Foo2(x, y) => println(s"Foo2($x, $y)")
    case Foo3(x, y) => println(s"Foo3($x, $y)")
    case Foo4(x, y) => println(s"Foo4($x, $y)")
